source/tomopy/prep/stripe.py

```diff
@@ -939,3 +941,156 @@ def _remove_stripe_based_interpolation(tomo, snr, size, drop_ratio, norm):
         sino = tomo[:, m, :]
         sino = _rs_interpolation(sino, snr, size, drop_ratio, norm)
         tomo[:, m, :] = sino
+
+
+def stripes_detect3d(arr, vert_filter_size_perc=5, radius_size=3, ncore=None):
+    """
+    Apply a stripes detection method to empasize their edges in a 3D array.
+    The input must be normalized projection data in range [0,1] and given in
+    the following axis orientation [angles, detY(depth), detX (horizontal)]. With 
+    this orientation, the stripes are the vertical features. The method works with
+    full and partial stripes of constant ot varying intensity. 
+
+    .. versionadded:: 1.13
+
+    Parameters
+    ----------
+    arr : ndarray
+        Input 3D array of float32 data type, normalized and in the following axis orientation [angles, detY(depth), detX (horizontal)].
+    vert_filter_size_perc : float, optional
+        The size (in percents relative to angular dimension) of the vertical 1D median filter to remove outliers.
+    radius_size : int, optional
+        The size of the filter to calculate the ratio. This will effect the width of the resulting mask.
+    ncore : int, optional
+        Number of cores that will be assigned to jobs. All cores will be used
+        if unspecified.
+
+    Returns
+    -------
+    ndarray
+        Weights for stripe's edges as a 3D array of float32 data type. 
+        The weights can be thresholded or passed to stripes_mask3d function to obtain a binary mask.
+
+    Raises
+    ------
+    ValueError
+        If the input array is not three dimensional.
+
+    """
+    if ncore is None:
+        ncore = mproc.mp.cpu_count()
+
+    input_type = arr.dtype
+    if (input_type != 'float32'):
+        arr = dtype.as_float32(arr)  # silent convertion to float32 data type
+    out = np.empty_like(arr, order='C')
+    
+    if arr.ndim == 3:
+        dz, dy, dx = arr.shape
+        if (dz == 0) or (dy == 0) or (dx == 0):
+            raise ValueError("The length of one of dimensions is equal to zero")
+    else:
+        raise ValueError("The input array must be a 3D array")
+
+    # calculate absolute values based on the provided percentages: 
+    if 0.0 < vert_filter_size_perc <= 100.0:
+        vertical_filter_size = (int)((0.01*vert_filter_size_perc)*dz)
+    else:
+        raise ValueError("vert_filter_size_perc value must be in (0, 100] percentage range ")
```

Comment:
Why is the Python API so different from the C API? Why not just expose `vertical_filter_size` directly? Accepting a float value that is cast down to an integer gives the illusion of greater precision?

Reply:
very good point. I've been thinking about this one. The reason why I went for more relative to the sizes of the data parameter is to avoid a problem of redefining this parameter every time when the sizes of data change. Also you need to know the sizes of the data to set it to something meaningful, while in precents you can  take something approximate and will work. 
I'm not 100% about this one, would you rather go for an absolute value?

Reply:
They are independent indeed, but the stripe's width can be relative to the total size of the detectors array. I guess for me it was easier to relate to the total size, e.g., 1% of the total detector size rather than to pick an exact absolute value in pixels. This choice is obviously subjective and I'm happy to change to absolute values. Obviously for absolute values you would certainly need to change them if the data dimensions change, they are not scalable.

Reply:
thought more about it and decided to revert to absolute values. Will commit the changes soon.